app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,39 @@
+class ApplicationsController < ApplicationController
+  def show 
+    @app = Application.find(params[:id]) 
+    @pets = @app.pets
+    if params[:search]
+      @found_pets = Pet.search(params[:search])
+    end
+  end 
```

Comment:
I noticed in your video that you had an error occur when you tried to visit an applications show page after creating an app and this is because of line 4. This should be nested inside a conditional because a new application would not have any pets to start with and therefore `@pets` would contain the value `nil` and you cannot iterate over `nil`.

```ruby
def show 
  @app = Application.find(params[:id]) 
  app_pets = @app.pets

  if app_pets # something like this could work
    @pets = app_pets
  end

  if params[:search]
    @found_pets = Pet.search(params[:search])
  end
end 
```

You would also need a conditional in your view to only iterate over `@pets` if `@pets` was not `nil`.

```erb
<% if @pets %>
  # Do the work...
<% end %>
```